spec/javascript/availability/components/ill_link.jsx.spec.js

```diff
@@ -0,0 +1,182 @@
+import {render, waitFor} from '@testing-library/react';
+import '@testing-library/jest-dom'
+import IllLink from '../../../../app/javascript/availability/components/ill_link';
+import availability from '../../../../app/javascript/availability';
+
+const baseUrl = 'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/' +
+                'OpenURL?Action=10';
+
+const testLink = async (getByRole, container, label, href) => {
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent(label);
+  expect(getByRole('link')).toHaveAttribute('href', href);
+};
+
+describe('when all fields are present', () => {
+  const moreParams = '&title=book%20title&callno=123&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu' +
+                     '&aulast=author%20name&date=2021';
+
+  beforeEach(() => {
+    $.get = jest.fn().mockImplementation(() => {
+      return Promise.resolve({
+        title_245ab_tsim: 'book title',
+        author_tsim: 'author name',
+        pub_date_illiad_ssm: 2021,
+        isbn_ssm: '1234'
+      });
+    });
+  });
+
+  test('renders an ILL link', async () => {
+    const {getByRole, container} = render(
+      <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+    );
+
+    const href = baseUrl + '&Form=30&isbn=1234' + moreParams;
+
+    await testLink(
+      getByRole,
+      container,
+      'Copy unavailable, request via Interlibrary Loan',
+      href
+    );
+  });
+  
+  test('renders a reserves scan link', async () => {
+    availability.reservesScanLocations = ['BINDERY'];
+  
+    const {getByRole, container} = render(
+      <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+    );
+
+    const href = baseUrl + 
+      '&Form=30&isbn=1234' +
+      '&Genre=GenericRequestReserves&location=BINDERY' + // reserves scan params
+      moreParams;
+
+    await testLink(
+      getByRole,
+      container,
+      'Copy unavailable, request via Interlibrary Loan',
+      href
+    );
+  
+    availability.reservesScanLocations = [];
```

Comment:
@nicktiberi is this for to clean this var for the rest of tests?

Reply:
Yep exactly. Though maybe I should store the previous value of `reservesScanLocations` and reset it, rather than setting it to an empty array at the end. This way if we add values back to `reservesScanLocations` it won't break stuff in the tests.

Reply:
it makes sense, just wanted to make sure if I am understanding correctly :) thanks!